.github/workflows/ci-test-suites.yml

```diff
@@ -28,14 +28,12 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
 # DAMAGE.
 
-# Github Actions workflow for Continuous Integration testing on Linux.
-# TODO i#4549: This is a work-in-progress and not enabled currently.
+# Github Actions workflow for Continuous Integration testing on Linux and MacOS.
```

Comment:
@derekbruening Note that all the test suites are part of one workflow. One drawback of this setup is that any re-run will start all the test suites again (unlike Travis where we could re-run individual suites), which seems a waste of resources. We could alternatively split this into 4 workflows, one for x86, all arm, macos, and clang-format.

Reply:
Is there any downside to having 4 workflows instead of 1?

Reply:
I couldn't find anything else other than code duplication, which wouldn't be that much anyway.

If there are multiple workflows for some event, all of them will be triggered on the event (https://github.community/t/execution-order-for-multiple-workflows-in-one-repo/116780). There's a limit on number of concurrent jobs (https://docs.github.com/en/free-pro-team@latest/actions/reference/usage-limits-billing-and-administration#usage-limits), but that will remain the same in both multiple or single workflow design.

Reply:
I've split it into multiple smaller workflows. PTAL if this seems fine.

Reply:
Perhaps the names of individual jobs can be simplified: 'osx-64bit' -> '64-bit', because the workflow name provides the remaining context. But may be better to keep them verbose, so that they make sense even out of context.

Reply:
I would keep the job name as is, but perhaps `s/ci-test-suite-/ci-/`, so `ci-osx / osx-64bit`.  "ci" kind of implies a test suite.  Or `s/ci-test-suite-/suite/` for `suite-osx / osx-64bit`.  Deployment jobs would probably not have 'ci" or "suite" in the name.

I would also rename the "x86" job to "x86-32".  Maybe "osx-x86-64" to match (the implied "linux-").

Reply:
Done. Thanks.